operator/v1/types_dns.go

```diff
@@ -48,6 +48,11 @@ type DNSSpec struct {
 	// +optional
 	Servers []Server `json:"servers,omitempty"`
 
+	// upstreamResolvers defines a schema for configuring CoreDNS
+	// to proxy DNS messages to upstream resolvers for the case of the
+	// default (".") server
+	UpstreamResolvers ForwardPlugin `json:"upstreamResolvers"`
```

Comment:
We need the default behavior to be the current behavior, which is to use `/etc/resolv.conf` with the "Sequential" policy.  Re-using the `ForwardPlugin` type has the potential to cause confusion: 
* Is the default configuration represented by leaving `spec.upstreamResolvers.upstreams` empty?  
* Is the default policy going to be "Sequential" when `spec.upstreamResolvers.upstreams` is empty and "Random" when `spec.upstreamResolvers.upstreams` is nonempty, or will it always be "Sequential"?  

Did you consider alternative designs?  A couple options come to mind:
* Define and use custom types with one field to specify the policy (with the default value "Sequential") and another field to specify a list of upstreams (with the default value empty, which would be explicitly documented to mean "use `/etc/resolv.conf`").  
* Allow a "default" entry with zone "." to be specified in `spec.forwardPlugin`, and special-case the "." entry in cluster-dns-operator.  

Those are just some suggestions.  Maybe the approach in this PR could work with some modifications, or maybe there is a better approach we haven't yet thought of.